src/metabase/pulse/render/body.clj

```diff
@@ -78,13 +78,28 @@
     :else
     (str value)))
 
+(defn- viz-settings-for-col
+  [col viz-settings]
+  (let [[_ field-id _]    (:field_ref col)
+        strip-ns          (fn [v] (if (map? v)
+                           (update-keys v #(keyword (name %)))
+                           v))
+        all-cols-settings (-> (update-keys viz-settings #(keyword (name %)))
+                              :column-settings
+                              (update-keys strip-ns)
+                              (update-keys #(into {} (take 2 %))) ;; throw out any unit/metadata to make key matching simpler
```

Comment:
> ;; throw out any unit/metadata to make key matching simpler

taking 2 on a map will get different keys depending on the map.

Reply:
This was done to turn something like this: `{:field-id 2 :metadata {:unit :default}}` into `{:field-id 2}`, but as you point out, that's unreliable since maps are not ordered, and should not be counted on to maintain any order.

So instead, I can update keys by selecting only those keys which would be used to match. For example:

```clojure
(-> viz-settings
    ::mb.viz/column-settings 
    (update-keys #(select-keys % [::mb.viz/field-id ::mb.viz/column-name])))
```